ddtrace/_trace/tracer.py

```diff
@@ -101,31 +87,23 @@ def _start_appsec_processor() -> Optional["AppSecSpanProcessor"]:
 
 def _default_span_processors_factory(
     trace_filters: List[TraceProcessor],
-    trace_writer: TraceWriter,
+    writer: Optional[TraceWriter],
     partial_flush_enabled: bool,
     partial_flush_min_spans: int,
-    compute_stats_enabled: bool,
-    single_span_sampling_rules: List[SpanSamplingRule],
-    agent_url: str,
-    trace_sampler: DatadogSampler,
     profiling_span_processor: EndpointCallCounterProcessor,
-) -> Tuple[List[SpanProcessor], Optional["AppSecSpanProcessor"], List[SpanProcessor]]:
-    # FIXME: type should be AppsecSpanProcessor but we have a cyclic import here
+) -> Tuple[List[SpanProcessor], Optional["AppSecSpanProcessor"], SpanAggregator]:
     """Construct the default list of span processors to use."""
     trace_processors: List[TraceProcessor] = []
     trace_processors += [
         PeerServiceProcessor(_ps_config),
         BaseServiceProcessor(),
-        TraceSamplingProcessor(
-            compute_stats_enabled, trace_sampler, single_span_sampling_rules, asm_config._apm_opt_out
-        ),
-        TraceTagsProcessor(),
     ]
     trace_processors += trace_filters
 
     span_processors: List[SpanProcessor] = []
     span_processors += [TopLevelSpanProcessor()]
 
+    appsec_processor: Optional["AppSecSpanProcessor"] = None
```

Comment:
It's not ideal that there's AppSec code in the Tracer class. Fixing that issue is well beyond the scope of this change, but just wanted to mention it.

Reply:
Yup the `tracer.configure(...)` should not be used to configure ASM/IAST. We should provide another interface for manually enabling these products. Once we define this new interface we can start removing appsec code from the tracing modules. 

cc: @avara1986